_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,669 @@
+
+#### Deploy the Microservice to Azure AKS
+
+You can manage a Kubernetes service purely with `kubectl`. However, there are some pretty helpful tools for monitoring and logging. Both [k9s](https://github.com/derailed/k9s) and [Kubernetes Lens](https://k8slens.dev/) are great. I recommend installing one or both of these and using them to inspect and monitor your Kubernetes services. They are especially helpful when things go wrong (not that things ever go wrong, I wouldn't know anything about that, I just heard about it from friends, I swear). Kubernetes Lens is a full-on desktop app that describes itself as a Kubernetes IDE. In comparison, k9s is a lighter-weight, text-based tool. 
+
+Open a Bash shell and navigate to the `k8s` subdirectory of the project.
+
+Run the microservice with: 
+
+```bash
+./kubectl-apply.sh -f
+```
+
+If you open this file, you'll see that it creates the namespace and applies the project files. If you do this manually, it's important that the namespace is created first and that the registry is run before the other services.
+
+```bash
+...
+suffix=k8s
+kubectl apply -f namespace.yml
+kubectl apply -f registry-${suffix}/
+kubectl apply -f blog-${suffix}/
+kubectl apply -f gateway-${suffix}/
+kubectl apply -f store-${suffix}/
+...
+```
+
+You can check on your pods with the following command. Everything was up and running almost immediately for me on AKS.
+
+```bash
+kubectl get pods -n demo
+```
+
+```bash
+NAME                                  READY   STATUS    RESTARTS   AGE
+blog-6896f6dd58-mbjkn                 1/1     Running   0          3m51s
+blog-neo4j-0                          1/1     Running   0          3m48s
+gateway-7f6d57765f-2fhfb              1/1     Running   0          3m46s
+gateway-postgresql-647476b4d5-jdp5c   1/1     Running   0          3m44s
+jhipster-registry-0                   1/1     Running   0          3m52s
+jhipster-registry-1                   1/1     Running   0          3m39s
+store-7889695569-k4wkv                1/1     Running   0          3m41s
+```
+
+Here's an example of what this looks like with Kubernetes Lens (the containers were still booting).
+
+{% img blog/jhipster-k8s-azure/lens-booting.png alt:"K8s Lens screenshot" width:"1000" %}{: .center-image }
+
+Another useful command is `describe`. I won't replicate the output here, but if you want more detailed information for debugging, you can also run the following.
+
+```bash
+kubectl describe pods -n demo
+```
+
+To tail logs, you can use the name of the pod.
+
+```
+kubectl logs <pod-name> --tail=-1 -n demo
+```
+
+Although, like I said, k9s and Lens are really the way to go for more detailed inspection.
+
+Here's a screenshot from k9s. You can dig down into the different pods to get more detailed information and inspect logs.
+
+{% img blog/jhipster-k8s-azure/k9s-screenshot.png alt:"Authenticated" width:"1000" %}{: .center-image }
```

Comment:
These last couple of screenshots are hard to read. I don't know if it's possible to bump the fonts and reshoot them. If not, no worries.